exonum/src/runtime/communication_channel.rs

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::runtime::{
+    dispatcher::Dispatcher,
+    mailbox::{Action, AfterRequestCompleted, BlockchainMailbox},
+    CallInfo, ConfigChange, ExecutionContext, ExecutionError, InstanceId,
+};
+
+/// Marker trait for `Supervisor` entities. Implementors of that trait can get access
+/// to the extended interface of `CommunicationChannel`.
+#[doc(hidden)]
+pub trait SupervisorAccess {}
+
+/// Communication channel is an proxy entity for performing service calls from
+/// other service instances.
+#[derive(Debug)]
+pub struct CommunicationChannel<'a, T = ()> {
+    pub(crate) mailbox: &'a BlockchainMailbox,
+    dispatcher: &'a Dispatcher,
+    phantom: std::marker::PhantomData<T>,
+}
+
+impl<'a, T> CommunicationChannel<'a, T> {
+    /// Creates a new communication channel.
+    pub(crate) fn new(mailbox: &'a BlockchainMailbox, dispatcher: &'a Dispatcher) -> Self {
+        Self {
+            mailbox,
+            dispatcher,
+            phantom: std::marker::PhantomData,
+        }
+    }
+
+    /// Call the corresponding runtime method.
```

Comment:
runtime or service method?

Reply:
Also, the PR description says the dispatcher is needed to call the other services — but there is also this method. Which is the right one to use?